presto/presto.go

```diff
@@ -1459,49 +1387,54 @@ func (c *typeConverter) ConvertValue(v interface{}) (driver.Value, error) {
 			return nil, err
 		}
 		return vv.Float64, err
-	case "date", "time", "time with time zone", "timestamp", "timestamp with time zone":
-		vv, err := scanNullTime(v)
-		if !vv.Valid {
-			return nil, err
-		}
-		return vv.Time, err
 	case "map":
 		if err := validateMap(v); err != nil {
 			return nil, err
 		}
 		return v, nil
-	case "array":
-		if err := validateSlice(v); err != nil {
-			return nil, err
-		}
-		return v, nil
-	case "row":
+	case "array", "row":
 		if err := validateSlice(v); err != nil {
 			return nil, err
 		}
 		return v, nil
 	default:
-		return nil, fmt.Errorf("type not supported: %q", c.typeName)
+		return v, nil
 	}
 }
 
 func validateMap(v interface{}) error {
 	if v == nil {
 		return nil
 	}
-	if _, ok := v.(map[string]interface{}); !ok {
+
+	str, ok := v.(string)
+	if !ok {
+		return fmt.Errorf("cannot convert %v (%T) to map", v, v)
+	}
+
+	var out map[string]interface{}
+	if err := json.Unmarshal([]byte(str), &out); err != nil {
```

Comment:
Presto returns maps, arrays, and rows as strings that contain serialized JSON objects. That differs from the Trino behavior, where they're returned as full-fledged JSON objects (not serialized into a string value). This change (and the similar change below) accounts for that difference.